src/ui/public/share/directives/share.js

```diff
@@ -1,18 +1,146 @@
+import {
+  getUnhashableStatesProvider,
+  unhashUrl,
+} from 'ui/state_management/state_hashing';
+import Notifier from 'ui/notify/notifier';
+
+import urlShortenerProvider from '../lib/url_shortener';
+
 import uiModules from 'ui/modules';
 import shareTemplate from 'ui/share/views/share.html';
 const app = uiModules.get('kibana');
 
-app.directive('share', function () {
+app.directive('share', function (Private) {
+  const getUnhashableStates = Private(getUnhashableStatesProvider);
+  const urlShortener = Private(urlShortenerProvider);
+
   return {
     restrict: 'E',
     scope: {
       objectType: '@',
       objectId: '@',
-      setAllowEmbed: '&?allowEmbed'
+      allowEmbed: '@',
     },
     template: shareTemplate,
-    controller: function ($scope) {
-      $scope.allowEmbed = $scope.setAllowEmbed ? $scope.setAllowEmbed() : true;
+    controllerAs: 'share',
+    controller: function ($scope, $location, globalState) {
+      if ($scope.allowEmbed !== 'false' && $scope.allowEmbed !== undefined) {
+        throw new Error('allowEmbed must be "false" or undefined');
+      }
+
+      // Default to allowing an embedded IFRAME, unless it's explicitly set to false.
+      this.allowEmbed = $scope.allowEmbed === 'false' ? false : true;
```

Comment:
Thanks @spalger , I see what you're saying. I prefer the way it is currently because I think it's clearer that `true` is the default value _unless_ "false" is passed in.